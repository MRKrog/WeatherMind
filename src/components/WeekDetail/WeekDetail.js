import React from 'react';
import PropTypes from 'prop-types';
import { findIcon } from '../../utility/findIcon';

export const WeekDetail = ({ time, tempHigh, tempLow, icon }) => {

  let iconImage = findIcon(icon);

  return (
    <div className="WeekDetail">
      <h5 className="todayDay">{time}</h5>
      <div className="weatherIcon">
        <img alt={icon} src={iconImage} />
      </div>
      <section className="tempLevels">
        <h6 className="tempHigh">{tempHigh}</h6>
        <h6 className="tempLow">{tempLow}</h6>
      </section>
    </div>
  )
}

WeekDetail.propTypes = {
  time: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  tempHigh: PropTypes.number.isRequired,
  tempLow: PropTypes.number.isRequired,
}
