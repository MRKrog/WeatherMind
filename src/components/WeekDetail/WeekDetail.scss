.WeekDetail {
  display: flex;
  align-items: center;
  justify-content: space-between;

  .todayDay {
    margin: 0;
    font-size: 1em;
    font-weight: 500;
  }

  .weatherIcon {
    width: 50px;
    height: 50px;

    .clear-night {
      width: 50px;
      height: 50px;
      display: block;
    }
  }

  .tempLevels {
    display: flex;

    h6 {
      font-size: 1em;
      font-weight: 500;
    }

    h6:first-child {
      padding-right: 1.5em;
    }

    .tempHigh {
      margin: 0;
    }

    .tempLow {
      margin: 0;
      color: #a5a5a5;
    }
  }
}
