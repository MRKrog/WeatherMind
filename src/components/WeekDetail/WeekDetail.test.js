import React from 'react';
import { shallow } from 'enzyme';
import { WeekDetail } from './WeekDetail';

describe('WeekDetail', () => {
  let wrapper;
  let mockIcon = "fog";
  let mockTempHigh = 45;
  let mockTempLow = 15;
  let mockTime = "04"

  beforeEach(() => {
    wrapper = shallow(
      <WeekDetail icon={mockIcon}
                  tempHigh={mockTempHigh}
                  tempLow={mockTempLow}
                  time={mockTime}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
