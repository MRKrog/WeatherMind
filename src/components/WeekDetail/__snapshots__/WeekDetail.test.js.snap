// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekDetail should match the snapshot 1`] = `
<div
  className="WeekDetail"
>
  <h5
    className="todayDay"
  >
    04
  </h5>
  <div
    className="weatherIcon"
  >
    <img
      alt="fog"
      src="fog.svg"
    />
  </div>
  <section
    className="tempLevels"
  >
    <h6
      className="tempHigh"
    >
      45
    </h6>
    <h6
      className="tempLow"
    >
      15
    </h6>
  </section>
</div>
`;
